Documentation/teaching/info/extra-vm.rst

```diff
@@ -0,0 +1,125 @@
+===================================
+Extending the Virtual Machine Setup
+===================================
+
+Connect to the Virtual Machine via SSH
+--------------------------------------
+
+The default Yocto image for the QEMU virtual machine
+(``core-image-minimal-qemu``) provides the minimal functionality to run the
+kernel and kernel modules. For extra features, such as an SSH connection,
+a more complete image is required, such as ``core-image-sato-dev-qemu``.
+
+To use the new image, update the ``YOCTO_IMAGE`` variable in
+``tools/labs/qemu/Makefile``:
+
+.. code-block::
+
+   YOCTO_IMAGE = core-image-sato-qemu$(ARCH).ext4
+
+When you start the virtual machine the first time using ``make boot`` with the
+new image configuration, it will download the image and then boot the virtual
+machine. The image is larger (around 400MB) than the minimal image so expect
+some time for the download.
+
+You then enter the virtual machine via ``minicom``, determine the IP address of
+the ``eth0`` interface an then you can connect to the virtual machine via SSH:
+
+.. code-block:: shell
+
+   $ mincom -D serial.pts
+   Poky (Yocto Project Reference Distro) 2.3 qemux86 /dev/hvc0
+
+   qemux86 login: root
+   root@qemux86:~# ip a s
+   1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
+       link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+       inet 127.0.0.1/8 scope host lo
+          valid_lft forever preferred_lft forever
+       inet6 ::1/128 scope host 
+          valid_lft forever preferred_lft forever
+   2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
+       link/ether 52:54:00:12:34:56 brd ff:ff:ff:ff:ff:ff
+       inet 172.213.0.18/24 brd 172.213.0.255 scope global eth0
+          valid_lft forever preferred_lft forever
+       inet6 fe80::5054:ff:fe12:3456/64 scope link 
+          valid_lft forever preferred_lft forever
+   3: sit0@NONE: <NOARP> mtu 1480 qdisc noop qlen 1000
+       link/sit 0.0.0.0 brd 0.0.0.0
+
+   $ ssh -l root 172.213.0.18
+   The authenticity of host '172.213.0.18 (172.213.0.18)' can't be established.
+   RSA key fingerprint is SHA256:JUWUcD7LdvURNcamoPePMhqEjFFtUNLAqO+TtzUiv5k.
+   Are you sure you want to continue connecting (yes/no)? yes
+   Warning: Permanently added '172.213.0.18' (RSA) to the list of known hosts.
+   root@qemux86:~# uname -a
+   Linux qemux86 4.19.0+ #3 SMP Sat Apr 4 22:45:18 EEST 2020 i686 GNU/Linux
+
+Connecting a Debugger to the Virtual Machine Kernel
```

Comment:
This one I prefer to keep it in the original page since it is used in a few lectures examples.